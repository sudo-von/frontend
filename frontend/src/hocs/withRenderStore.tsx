import { render } from '@testing-library/react';
import { ReactNode } from 'react';
import { ThemeStore } from 'src/contexts/ThemeContext/ThemeContext';

const withRenderStore = (component: ReactNode) => {
  const WithRenderStore = () => {
    return <ThemeStore>{component}</ThemeStore>;
  };
  return render(<WithRenderStore />);
};

export default withRenderStore;
