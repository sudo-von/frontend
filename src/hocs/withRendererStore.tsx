import renderer from 'react-test-renderer';
import { ReactNode } from 'react';
import { ThemeStore } from 'src/contexts/ThemeContext/ThemeContext';

const withRendererStore = (component: ReactNode) => {
  const WithRendererStore = () => {
    return <ThemeStore>{component}</ThemeStore>;
  };
  return renderer.create(<WithRendererStore />).toJSON();
};

export default withRendererStore;
