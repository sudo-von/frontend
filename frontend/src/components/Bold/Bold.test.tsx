import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import withRendererStore from 'src/hocs/withRendererStore';
import withRenderStore from 'src/hocs/withRenderStore';
import Bold from './Bold';

describe('Bold component', () => {
  test('Should display a bold text', () => {
    withRenderStore(<Bold>Hello world</Bold>);
    expect(screen.getByText('Hello world')).toHaveStyle('font-weight: bold');
  });

  test('Should render a bold text', () => {
    const tree = withRendererStore(<Bold>Hello world</Bold>);
    expect(tree).toMatchSnapshot();
  });
});
