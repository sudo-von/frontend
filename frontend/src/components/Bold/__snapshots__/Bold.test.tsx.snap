// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bold component Should render a bold text 1`] = `
<span
  className="sc-bdvvtL fhkyxG"
>
  Hello world
</span>
`;
